renderer/modules/shared/components/drawer/navigation/custom-drawer/custom-drawer.tsx

```diff
@@ -1,17 +1,103 @@
+import { useState, useEffect } from 'react';
 import { useRouter } from 'next/router';
 import { drawerLinks, DrawerLink } from './drawer-links';
 import List from '@material-ui/core/List';
 import SectionLink from '../section-link/section-link';
 import useDawerStyles from '../../drawer.style';
-import UpgradeBanner from '../../../upgrade-banner/upgrade-banner';
+import UpgradeBanner, {
+  UpgradeBannerProps,
+} from '../../../upgrade-banner/upgrade-banner';
 import ProfilePicture from '../../../profile-picture/profile-picture';
+import { IpcRendererEvent } from 'electron';
 
 interface CustomDrawerProps {
   classes: ReturnType<typeof useDawerStyles>;
 }
 
+interface IdeDistribution {
+  id: string;
+  ideConfig: {
+    basepath: string;
+    workspaces: string;
+    commands: string;
+    version: string;
+  };
+}
+
+interface DevonIdeScripts {
+  version: string;
+  updated: Date;
+}
 export default function CustomDrawer(props: CustomDrawerProps): JSX.Element {
   const router = useRouter();
+  const initialProps: UpgradeBannerProps = {
+    version: '',
+    infoText: '',
+  };
+  const [displayUpgradeBanner, setDisplayUpgradeBanner] = useState(false);
+  const [upgradeBannerProps, setUpgradeBannerProps] = useState<
+    UpgradeBannerProps
+  >(initialProps);
+
+  const getLocalLatestVersion = (versions: string[]): string => {
+    const numericVersions = versions.map((v) => {
+      return +v.split('.').join(''); // for input "2020.04.001" it will return 202004001
+    });
+    const maxVersion = Math.max(...numericVersions); // for input [202008001, 202004002, 202004004] it will return 202008001
+    const latestLocalVersion = versions.filter(
+      (v) => +v.split('.').join('') === maxVersion
+    );
```

Comment:
Here you won't need to do the split join (check previous comment).
Use `versions.find()` instead. This way you won't need `latestLocalVersion[0]`
Also be careful: What does happen when versions is an empty array? (I don't know if that case can happen)

Reply:
Yes `versions.find()` can be used instead.
I don't think `versions` will be an empty array, since the function in which it is residing will be called only in the presence of some version data.